lib/matplotlib/cbook.py

```diff
@@ -2331,42 +2331,56 @@ def _picklable_class_constructor(mixin_class, fmt, attr_name, base_class):
 
 
 def _is_torch_array(x):
-    """Check if 'x' is a PyTorch Tensor."""
+    """Return whether *x* is a PyTorch Tensor."""
     try:
-        # we're intentionally not attempting to import torch. If somebody
-        # has created a torch array, torch should already be in sys.modules
-        return isinstance(x, sys.modules['torch'].Tensor)
-    except Exception:  # TypeError, KeyError, AttributeError, maybe others?
-        # we're attempting to access attributes on imported modules which
-        # may have arbitrary user code, so we deliberately catch all exceptions
-        return False
+        # We're intentionally not attempting to import torch. If somebody
+        # has created a torch array, torch should already be in sys.modules.
+        tp = sys.modules.get("torch").Tensor
+    except AttributeError:
+        return False  # Module not imported or a nonstandard module with no Tensor attr.
+    return (isinstance(tp, type)  # Just in case it's a very nonstandard module.
```

Comment:
When would that happen? `tp` should be `torch.Tensor` and I doubt that `isinstance(torch.Tensor, type)` will ever change. In fact, if we'd do a regular `import torch` we would also just do the standard isinstance check.

Reply:
Ok.